pkg/agent/client.go

```diff
@@ -80,6 +80,7 @@ type connectionManager struct {
 func (cm *connectionManager) Add(connID int64, ctx *connContext) {
 	cm.mu.Lock()
 	defer cm.mu.Unlock()
+	metrics.Metrics.EndpointConnectionInc()
```

Comment:
Inc/dec everytime on Metrics *should* get you the correct answer, as long as no one changes the map outside of these methods. However we can also just use the  map length.

Reply:
There isn't a clientset map length, only one per client. There is some discussion here on this, maybe buried.